### PR TITLE
feat: use php-http-discovery and remove concrete guzzle

### PR DESCRIPTION
**Pull request by mikelovely**

### CODE REVIEW

**Review by stevehill1981 — APPROVED** [review]
This looks like a nice change.

I can't spot anything obviously wrong with the code, and the tests still pass.

File: lib/Wrappers/HttpWrapper.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     * @param string $baseUrl
+     * @param string $apiKey
+     * @param ClientInterface $httpClient
+     * @param RequestFactory $requestFactory
+     */
+    public function __construct(
+        string $baseUrl,
+        string $apiKey,
+        ClientInterface $httpClient = null,
+        RequestFactory $requestFactory = null
```
Comment (nunocosta-divido) [inline lib/Wrappers/HttpWrapper.php:47]: should we make the types as nullable too ? eg: `? ClientInterface` and `? RequestFactory`

Comment (mikelovely) [reply lib/Wrappers/HttpWrapper.php:47]: It needs to be 5.6 compatible.

**Review by nunocosta-divido — APPROVED** [review]

**Review by andivido — APPROVED** [review]
All plugins affected by this are now locked to the current version of the sdk. We should be good to make this change now (at least internally)

**Pull request merged**